docs/conf.py

```diff
@@ -527,7 +557,72 @@ def _parse_confval_signature(
     return signature
 
 
+class TestColor(SphinxRole):
+    color_type: str
+    style = (
+        "background-color: %s;"
```

Comment:
Can we get this from a CSS var so that we don't have to duplicate the colors used by the theme in `supported_colors`?

Reply:
The palette names are only used in CSS selectors. It may be inefficient to crawl the CSS bundle for the supported palette colors.

On a side note, I've been working up a `social_cards` ext that should mimic the upstream's [builtin social plugin](https://squidfunk.github.io/mkdocs-material/setup/setting-up-social-cards/#+social.cards_layout_options). Coincidentally, the list of [supported palette colors are saved in the python source for the plugin](https://github.com/squidfunk/mkdocs-material/blob/1b3b8a719db8209c2bd9e6ee9ddaef5c78420fc7/src/plugins/social/plugin.py#L497-L519). So, this dict will be migrated out of conf.py when I have something ready to submit.

Reply:
I found that the color names can be extracted from the css using the following script:

```python
import pathlib
import re


def get_colors(t):
    unique_colors = {m.group(1) for m in re.finditer(
        r"\[data-md-color-" + t + r"=([a-z\-]+)\]",
        pathlib.Path("sphinx_immaterial/bundles/stylesheets/palette.css").read_text(
            encoding="utf-8"
        ),
    )}
    return sorted(unique_colors)


print(f'Primary: {get_colors("primary")}')
print(f'Accent: {get_colors("accent")}')
```

Getting the color values corresponding to these names from the CSS might be kind of annoying, but I think there is a workaround: set the `data-md-color-primary` or `data-md-color-accent` attribute on the `button` element; then you can access the color values from other style rules on the button via css variables.

Reply:
> set the data-md-color-primary or data-md-color-accent attribute on the button element

Yep, I was going to do that also, but I got hung up writing #261 proposal. I'll also play around with the other inlined CSS to see what can go or is still required.

Reply:
As for the scripting idea, I think it is kinda moot since I still need the color data for the social_cards ext. I can add it as a temp workaround for now. For the social_cards ext, I'll also try expanding on your snippet to extract the color data that corresponds to the color name.

Reply:
done. I added a lambda to sort the colors a little better, but it still looks a bit helter-skelter:
![image](https://github.com/jbms/sphinx-immaterial/assets/14963867/e4a8a5ae-6fc7-4f9a-8eb4-4cdc879c6ee9)